food-truck-api/src/main/java/food/truck/api/routes/RouteLocation.java

```diff
@@ -36,4 +37,22 @@
     @Convert(converter = PositionConverter.class)
     @Column(nullable = false)
     Position position;
+
+    @JsonIgnore
+    public boolean wrapsAroundMidnight() {
+        return exitTime.isBefore(arrivalTime);
+    }
+
+    // I have no idea how I came up with this but the tests seem to work
+    // 4 cases, depending on which locations wrap around midnight (UTC)
+    public boolean timeConflictsWith(RouteLocation other) {
+        if (!wrapsAroundMidnight() && !other.wrapsAroundMidnight())
+            return !arrivalTime.isAfter(other.exitTime) && !exitTime.isBefore(other.arrivalTime);
+        else if (wrapsAroundMidnight() && !other.wrapsAroundMidnight())
```

Comment:
The code on these two 'one wraps around at midnight' branches are identical, just flipped around the `||`.